### DISCUSSION TITLE
Enhancement: Restriction of access based on Google group membership

### DISCUSSION THREAD

**Discussion by cablespaghetti** [danny-avila/LibreChat#1746] [Q&A]
### What features would you like to see added?

Currently using Google OIDC anyone in our organisation can login to LibreChat. Is there a way to allow restriction based on Google Group membership, or maybe allow passthrough of user information from a tool like https://github.com/oauth2-proxy/oauth2-proxy?

### More details

We would like to restrict down access to certain groups of people within our Google Workspace.

### Which components are impacted by your request?

General

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ndrsfel** [comment]
Sounds like #1488 would help?

**Reply by cablespaghetti** [reply]
I'm not certain Google specifically works this way but I'll have to do some research how Oauth2 Proxy achieves this functionality.

**Comment by owengo** [comment]
Hello, I have submitted https://github.com/danny-avila/LibreChat/pull/4613 , which allows to configure a group from your workspace account. Only the members ( direct or indirect ) of the groups are then allowed to login to librechat via google login.
